docs/concepts/results.md

```diff
@@ -193,3 +193,191 @@ def my_flow():
 
 my_flow()
 ```
+
+## Persisting results
+
+The Prefect API does not store your results [except in special cases](#storage-of-results-in-prefect). Instead, the result is _persisted_ to a storage location in your infrastructure and Prefect stores a _reference_ to the result.
+
+The following Prefect features require results to be persisted:
+
+- Task cache keys
+- Flow run retries
+
+If results are not persisted, these features will not be usable.
+
+### Configuring results
+
+Persistence of results requires a [**serializer**](#result-serializers) and a [**storage** location](#result-storage). Prefect sets defaults for these, and you should not need to adjust them until you want to customize behavior. You can configure results on the `flow` and `task` decorators with the following options:
+
+- `persist_result`: Whether the result should be persisted to storage.
+- `result_storage`: Where to store the result when persisted.
+- `result_serializer`: How to convert the result to a storable form.
+
+#### Toggling persistence
+
+Persistence of results can be configured with the `persist_result` option. The `persist_result` option defaults to a null value, which will automatically enable persistence if it is needed for a Prefect feature used by the flow or task. Persistence of results can be manually toggled on or off:
+
+```python
+from prefect import flow, task
+
+@flow(persist_result=True)
```

Comment:
For flows, `persist_result` makes me think that this is only for the (single) result of the flow, rather than a setting that will be inherited by all tasks (and subflows?) inside the flow. `persist_results` (plural) is what I would expect to see there. I realize this would break symmetry with the task decorator, but is a more accurate description of this behavior.

Reply:
It only affects the flow. Unlike the other options, this setting will not be inherited. How can I clarify this in the persistence toggle doc?

A separate setting i.e. `persist_child_results` or `persist_task_results` may be added in the future to toggle the default for children, but Bill and I decided it was not necessary in the first go.

Reply:
> Persistence of results can be configured with the `persist_result` option. The `persist_result` option defaults to a null value, which will automatically enable persistence if it is needed for a Prefect feature used by the flow or task.

vs 

> [The result storage location](#result-storage-types) can be configured with the `result_storage` option. The `result_storage` option defaults to a null value, which infers storage from the context.
Generally, this means that tasks will use the result storage configured on the flow unless otherwise specified.

Reply:
Sorry, I think I misunderstood -- I guess I assumed this was inherited because the other options were. In that case I retract my comment! I'm not sure if my confusion is universal (it could have just been me reading quickly) so I'm not sure if requires explicit documentation at this time.

Reply:
`persist_results` in the plural form is actually not bad and works for both -- 1) what if the task returns a tuple 2) one could see it as _persist results of multiple runs_ for this task/flow